lib/Optimizer/Transforms/ObserveAnsatz.cpp

```diff
@@ -174,9 +205,13 @@ struct AppendMeasurements : public OpRewritePattern<func::FuncOp> {
         qubitsToMeasure.push_back(qubitVal);
     }
 
-    for (auto &qubitToMeasure : qubitsToMeasure) {
+    for (auto &[measureNum, qubitToMeasure] :
+         llvm::enumerate(qubitsToMeasure)) {
       // add the measure
-      builder.create<quake::MzOp>(loc, builder.getI1Type(), qubitToMeasure);
+      char regName[16];
+      std::snprintf(regName, sizeof(regName), "r%05lu", measureNum);
```

Comment:
I didn't see anything in StringExtras.h for this. Maybe I'm not looking in the right place though.

Reply:
Are you suggesting that I change anything here?

Reply:
No. It just seemed like something that would already have a helper function in LLVM...